build error with HPX_WITH_NETWORKING=OFF

from a clean build dir
```
cmake -DHPX_WITH_CXX17=ON -DHPX_WITH_CUDA_SUPPORT=ON -DHPX_WITH_MALLOC=JEMALLOC -DHPX_WITH_NETWORKING=OFF ~/src/hpx/
```
barrier node appears to need distributed runtime (or some other feature that is not enabled)
```
[1/1] Building CXX object libs/collectives/CMakeFiles/hpx_collectives.dir/src/detail/barrier_node.cpp.o
FAILED: libs/collectives/CMakeFiles/hpx_collectives.dir/src/detail/barrier_node.cpp.o 
/usr/bin/g++  -DBOOST_DISABLE_ASSERTS -DHPX_DISABLE_ASSERTS -DHPX_EXPORTS -D_GNU_SOURCE -I/home/biddisco/src/hpx-branches/cuda-futures/libs/collectives/include -Ilibs/collectives/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/collectives/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/collectives/../.. -I/home/biddisco/src/hpx-branches/cuda-futures/libs/affinity/include -Ilibs/affinity/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/affinity/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/assertion/include -Ilibs/assertion/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/assertion/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/config/include -Ilibs/config/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/preprocessor/include -Ilibs/preprocessor/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/preprocessor/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/config_registry/include -Ilibs/config_registry/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/datastructures/include -Ilibs/datastructures/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/datastructures/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/concepts/include -Ilibs/concepts/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/concepts/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/type_support/include -Ilibs/type_support/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/type_support/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/errors/include -Ilibs/errors/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/errors/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/filesystem/include -Ilibs/filesystem/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/format/include -Ilibs/format/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/format/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/logging/include -Ilibs/logging/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/logging/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/timing/include -Ilibs/timing/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/timing/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/hardware/include -Ilibs/hardware/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/hardware/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/thread_support/include -Ilibs/thread_support/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/thread_support/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/topology/include -Ilibs/topology/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/topology/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/concurrency/include -Ilibs/concurrency/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/concurrency/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/basic_execution/include -Ilibs/basic_execution/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/basic_execution/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/hashing/include -Ilibs/hashing/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/hashing/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/util/include -Ilibs/util/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/util/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/iterator_support/include -Ilibs/iterator_support/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/iterator_support/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/functional/include -Ilibs/functional/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/functional/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/debugging/include -Ilibs/debugging/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/debugging/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/serialization/include -Ilibs/serialization/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/serialization/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/allocator_support/include -Ilibs/allocator_support/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/allocator_support/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/async/include -Ilibs/async/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/async/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/coroutines/include -Ilibs/coroutines/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/coroutines/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/execution/include -Ilibs/execution/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/execution/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/async_base/include -Ilibs/async_base/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/async_base/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/async_combinators/include -Ilibs/async_combinators/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/async_combinators/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/memory/include -Ilibs/memory/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/threading/include -Ilibs/threading/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/threading/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/futures/include -Ilibs/futures/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/futures/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/synchronization/include -Ilibs/synchronization/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/synchronization/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/threading_base/include -Ilibs/threading_base/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/threading_base/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/naming_base/include -Ilibs/naming_base/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/resource_partitioner/include -Ilibs/resource_partitioner/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/resource_partitioner/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/cache/include -Ilibs/cache/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/cache/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/command_line_handling/include -Ilibs/command_line_handling/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/command_line_handling/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/asio/include -Ilibs/asio/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/asio/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/batch_environments/include -Ilibs/batch_environments/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/batch_environments/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/string_util/include -Ilibs/string_util/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/mpi_base/include -Ilibs/mpi_base/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/mpi_base/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/runtime_configuration/include -Ilibs/runtime_configuration/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/runtime_configuration/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/plugin/include -Ilibs/plugin/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/plugin/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/prefix/include -Ilibs/prefix/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/prefix/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/version/include -Ilibs/version/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/program_options/include -Ilibs/program_options/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/pack_traversal/include -Ilibs/pack_traversal/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/pack_traversal/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/thread_pools/include -Ilibs/thread_pools/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/thread_pools/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/schedulers/include -Ilibs/schedulers/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/schedulers/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/threadmanager/include -Ilibs/threadmanager/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/threadmanager/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/io_service/include -Ilibs/io_service/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/io_service/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/thread_executors/include -Ilibs/thread_executors/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/thread_executors/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/executors/include -Ilibs/executors/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/executors/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/static_reinit/include -Ilibs/static_reinit/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/static_reinit/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/timed_execution/include -Ilibs/timed_execution/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/timed_execution/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/local_async/include -Ilibs/local_async/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/local_lcos/include -Ilibs/local_lcos/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/local_lcos/include_compatibility -I/home/biddisco/src/hpx-branches/cuda-futures/libs/performance_counters/include -Ilibs/performance_counters/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/statistics/include -Ilibs/statistics/include -I/home/biddisco/src/hpx-branches/cuda-futures/libs/statistics/include_compatibility -isystem /home/biddisco/spack/linux-pop20-skylake/gcc-8.4.0/boost-1.73.0-buzduqrwp4n3hj5rjxh5obm4d5nx72f5/include -isystem /home/biddisco/spack/linux-pop20-skylake/gcc-8.4.0/hwloc-1.11.11-3wn3v46mwp3qg6o6qbdz6fqbzsofr62s/include -isystem /home/biddisco/spack/linux-pop20-skylake/gcc-8.4.0/jemalloc-5.2.1-2c3qvekv7yvqwvf2b5bblbucn5gwtxva/include -fPIC -march=native -mtune=native -ffast-math -O3 -DNDEBUG -fPIC   -pthread -Wall -Wextra -Wno-unused-local-typedefs -Wno-strict-aliasing -Wno-sign-promo -Wno-attributes -Wno-cast-align -Wno-noexcept-type -Wno-unused-parameter -Wformat=2 -Wno-format-nonliteral -Winit-self -Wdouble-promotion -Wcast-qual -Wcast-align -Werror=trampolines -Werror=parentheses -Werror=reorder -Werror=return-type -Werror=sequence-point -Werror=uninitialized -Werror=format -Werror=missing-braces -Werror=sign-compare -fdiagnostics-show-option -Werror=vla -Wno-unused-but-set-parameter -Wno-unused-but-set-variable -Wno-maybe-uninitialized -Wno-sync-nand -Wno-delete-non-virtual-dtor -fvisibility=hidden -std=c++17 -MD -MT libs/collectives/CMakeFiles/hpx_collectives.dir/src/detail/barrier_node.cpp.o -MF libs/collectives/CMakeFiles/hpx_collectives.dir/src/detail/barrier_node.cpp.o.d -o libs/collectives/CMakeFiles/hpx_collectives.dir/src/detail/barrier_node.cpp.o -c /home/biddisco/src/hpx-branches/cuda-futures/libs/collectives/src/detail/barrier_node.cpp
In file included from /home/biddisco/src/hpx-branches/cuda-futures/libs/collectives/../../hpx/runtime/components/make_client.hpp:10,
                 from /home/biddisco/src/hpx-branches/cuda-futures/libs/collectives/../../hpx/runtime/basename_registration_fwd.hpp:15,
                 from /home/biddisco/src/hpx-branches/cuda-futures/libs/collectives/../../hpx/runtime_fwd.hpp:28,
                 from /home/biddisco/src/hpx-branches/cuda-futures/libs/async/include/hpx/async/applier/apply_helper.hpp:15,
                 from /home/biddisco/src/hpx-branches/cuda-futures/libs/async/include/hpx/async/applier/apply.hpp:12,
                 from /home/biddisco/src/hpx-branches/cuda-futures/libs/async/include/hpx/async/apply.hpp:10,
                 from /home/biddisco/src/hpx-branches/cuda-futures/libs/async/include/hpx/async.hpp:9,
                 from /home/biddisco/src/hpx-branches/cuda-futures/libs/collectives/src/detail/barrier_node.cpp:8:
/home/biddisco/src/hpx-branches/cuda-futures/libs/futures/include/hpx/futures/future.hpp: In instantiation of ‘static decltype(auto) hpx::lcos::detail::future_then_dispatch<Future, FD, Enable>::call(Future&&, T0&&, F&&) [with T0 = const hpx::detail::sync_policy&; F = hpx::lcos::detail::barrier_node::gather()::<lambda(hpx::lcos::future<void>)>; Future = hpx::lcos::future<std::vector<hpx::lcos::future<void> > >; FD = hpx::detail::sync_policy; Enable = void]’:
/home/biddisco/src/hpx-branches/cuda-futures/libs/futures/include/hpx/futures/future.hpp:649:74:   required by substitution of ‘template<class F, class T0> static decltype (hpx::lcos::detail::future_base<hpx::lcos::future<std::vector<hpx::lcos::future<void> > >, std::vector<hpx::lcos::future<void> > >::future_then_dispatch<typename std::decay<F>::type>::call(std::move(fut), forward<T0>(t0), forward<F>(f))) hpx::lcos::detail::future_base<hpx::lcos::future<std::vector<hpx::lcos::future<void> > >, std::vector<hpx::lcos::future<void> > >::then<F, T0>(hpx::lcos::future<std::vector<hpx::lcos::future<void> > >&&, T0&&, F&&, hpx::error_code&) [with F = hpx::lcos::detail::barrier_node::gather()::<lambda(hpx::lcos::future<void>)>; T0 = const hpx::detail::sync_policy&]’
/home/biddisco/src/hpx-branches/cuda-futures/libs/futures/include/hpx/futures/future.hpp:971:35:   required from ‘decltype(auto) hpx::lcos::future<R>::then(T0&&, F&&, hpx::error_code&) [with T0 = const hpx::detail::sync_policy&; F = hpx::lcos::detail::barrier_node::gather()::<lambda(hpx::lcos::future<void>)>; R = std::vector<hpx::lcos::future<void> >]’
/home/biddisco/src/hpx-branches/cuda-futures/libs/collectives/src/detail/barrier_node.cpp:204:14:   required from here
/home/biddisco/src/hpx-branches/cuda-futures/libs/futures/include/hpx/futures/future.hpp:426:42: error: static assertion failed: Cannot use the                     dummy implementation of future_then_dispatch, please use                     one of the template specialization.
  426 |             static_assert(sizeof(Future) == 0, "Cannot use the \
      |                           ~~~~~~~~~~~~~~~^~~~

```
and so on ...

--- a/libs/collectives/include/hpx/collectives/detail/communicator.hpp
+++ b/libs/collectives/include/hpx/collectives/detail/communicator.hpp
@@ -13,6 +13,7 @@
 #include <hpx/assertion.hpp>
 #include <hpx/async_base/launch_policy.hpp>
 #include <hpx/basic_execution/register_locks.hpp>
+#include <hpx/execution.hpp>
 #include <hpx/functional/bind_back.hpp>
 #include <hpx/futures/future.hpp>
 #include <hpx/local_lcos/and_gate.hpp>

--- a/libs/collectives/src/detail/barrier_node.cpp
+++ b/libs/collectives/src/detail/barrier_node.cpp
@@ -9,6 +9,7 @@
 #include <hpx/async_base/launch_policy.hpp>
 #include <hpx/async_combinators/when_all.hpp>
 #include <hpx/collectives/detail/barrier_node.hpp>
+#include <hpx/execution.hpp>
 #include <hpx/futures/future.hpp>
 #include <hpx/local_lcos/promise.hpp>
 #include <hpx/memory/intrusive_ptr.hpp>

--- a/libs/performance_counters/src/counters.cpp
+++ b/libs/performance_counters/src/counters.cpp
@@ -8,6 +8,7 @@
 #include <hpx/config.hpp>
 #include <hpx/assertion.hpp>
 #include <hpx/async/applier/apply.hpp>
+#include <hpx/execution.hpp>
 #include <hpx/format.hpp>
 #include <hpx/functional/bind_front.hpp>
 #include <hpx/functional/function.hpp>

--- a/src/runtime/agas/addressing_service.cpp
+++ b/src/runtime/agas/addressing_service.cpp
@@ -15,6 +15,7 @@
 #include <hpx/async.hpp>
 #include <hpx/basic_execution/register_locks.hpp>
 #include <hpx/errors.hpp>
+#include <hpx/execution.hpp>
 #include <hpx/format.hpp>
 #include <hpx/functional/bind.hpp>
 #include <hpx/functional/bind_back.hpp>

